geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -543,68 +542,61 @@ class AccumuloDataStore(val connector: Connector,
   /**
    * Computes and writes the metadata for this feature type
    */
-  private def writeMetadata(sft: SimpleFeatureType,
-                            fe: SerializationType,
-                            spatioTemporalSchemaValue: String) {
-
+  private def writeMetadata(sft: SimpleFeatureType) {
     // compute the metadata values
     val attributesValue             = SimpleFeatureTypes.encodeType(sft)
     val dtgValue: Option[String]    = sft.getDtgField // this will have already been checked and set
-    val featureEncodingValue        = /*_*/fe.toString/*_*/
+    val z2TableValue                = Z2Table.formatTableName(catalogTable, sft)
     val z3TableValue                = Z3Table.formatTableName(catalogTable, sft)
-    val spatioTemporalIdxTableValue = SpatioTemporalTable.formatTableName(catalogTable, sft)
     val attrIdxTableValue           = AttributeTable.formatTableName(catalogTable, sft)
     val recordTableValue            = RecordTable.formatTableName(catalogTable, sft)
     val queriesTableValue           = formatQueriesTableName(catalogTable)
     val tableSharingValue           = sft.isTableSharing.toString
     val enabledTablesValue          = sft.getEnabledTables.toString
-    val dataStoreVersion            = CURRENT_SCHEMA_VERSION.toString
+    val dataStoreVersion            = sft.getSchemaVersion.toString
+
+    // only set spatio-temporal fields if z2 isn't supported
+    val (stSchema, stTable) = if (sft.getSchemaVersion < 8) {
```

Comment:
Wondering if we can also avoid writing it if enabledTables does not contain st idx